### ISSUE TITLE
Expressing Interval / Extent  (e.g., Gannt Chart / Area Band)

### ISSUE THREAD

**Issue by kanitw** [vega/vega-lite#686]
Expressing Interval such as Gant Chart / Area Band requires specifying both `x` and `x2`.  

- [ ] design syntax
- [ ] find example data
- [ ] implement

**Comment by kanitw** [comment]
From discussion with @arvind, it might be sufficient to have `x` and `x2`   (and not having `width` for our marks).   (Same for y)

A few alternatives:
- Explicitly adding `x2` channels. 
- Make `x` an array, but that might be interfering with "x-repeat" semantic in extended unit spec (Tableau style spec).   (In that world, array of `x` means repeat x over these fields.)

I'm leaning toward adding `x2` `y2` channels for bar and area.  But we need to think through whether we have a case that we really need width or or height.   I think in general for expanding bar and area,  having `(x, x2)` is more useful than `(x,width)`.  Moreover, for expressiveness we can always add a formula to calculate the width.

**Comment by jheer** [comment]
Both of the alternatives sound reasonable to me, I don't have a strong opinion either way. I assume these additional options would only apply to bar and area marks?

**Comment by kanitw** [comment]
@jheer  exactly.   :)

**Comment by kanitw** [comment]
Skimming at our implementation, we actually output `(x,x2)` pair when x is a measure for bar and area.  
So this pairing kind make sense (same for y).

Another note of this is that if x2 is introduced, stacking no longer make sense.  So in this case, it would produce layered bar/stack chart with x2 set (for each group of data).

For example,

```
{
  "mark": "area",
  "encoding": {
    "y": {"field": "profit", "aggregate": "min", "type": "quantitative"},
    "y2": {"field": "profit", "aggregate": "max", "type": "quantitative"},
    "x": {"field": "time", "type": "temporal", "timeUnit": "year"},
    "color": {"field": "region", "type": "nominal"}
  }
}
```
It probably makes no sense to stack these min/max area bands.

**Comment by kanitw** [comment]
@ZeningQu  This is the issue I was talking about yesterday.  

Before we decide on `x2` or `width` (and `y2` or `height`) it might be worth surveying other tools (e.g., ggplot2, matlab, tableau to see how to they express this)

**Comment by kanitw** [comment]
Fixed in #1401 that will get merged soon.
